StickyHeaders/src/main/java/io/doist/recyclerviewext/sticky_headers/StickyHeadersLinearLayoutManager.java

```diff
@@ -296,6 +370,22 @@ protected void bindStickyHeader(@NonNull RecyclerView.Recycler recycler, int pos
         recycler.bindViewToPosition(mStickyHeader, position);
         mStickyHeaderPosition = position;
         measureAndLayoutStickyHeader();
+
+        // If we have a pending scroll wait until the end of layout and scroll again.
+        if (mPendingScrollPosition != RecyclerView.NO_POSITION) {
+            final ViewTreeObserver vto = mStickyHeader.getViewTreeObserver();
+            vto.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    vto.removeOnGlobalLayoutListener(this);
+
+                    if (mPendingScrollPosition != RecyclerView.NO_POSITION) {
+                        scrollToPositionWithOffset(mPendingScrollPosition, mPendingScrollOffset);
+                        setPendingScroll(RecyclerView.NO_POSITION, 0);
+                    }
+                }
+            });
+        }
```

Comment:
That was my expectation too. But scrolling is done in `LinearLayoutManager#onLayoutChildren` and this happens afterwards. So basically we trigger a layout during a layout and it's discarded (`RecyclerView#eatRequestLayout` :smile:). As a workaround, calling `mStickyHeader.requestLayout()` works on 5.0, but not on 4.1.

Looking at it again, we should probably pass the `layout` arg from `updateStickyHeader` here and if in a layout we could scroll right away. But if not we'd still have to add the listener.